src/python/s2_common.i

```diff
@@ -71,6 +107,15 @@
 
 %apply S2CellId *OUTPUT_ARRAY_4 {S2CellId neighbors[4]};
 
+%typemap(in, numinputs=0) S2Error * (S2Error err) {
```

Comment:
What does this do / where is it used? Maybe add a comment since it's not obvious.

Reply:
This pair of typemaps work together to translate errors from the C++ library into Python exceptions. The first declares a local variable `S2Error err;` and sets the matched argument (`$1`, of type `S2Error *`) to point to it; the second inspects the value of the `S2Error` after the call and throws an exception if its `ok()` method returns false. `S2BuilderTest.testException` in `pywraps2_test.py` tests this.

https://www.swig.org/Doc1.3/Typemaps.html#Typemaps_nn23 describes the syntax being used here.

I've added a comment.